Libplanet/Net/Swarm.cs

```diff
@@ -1597,29 +1597,43 @@ CancellationToken cancellationToken
 
             if (tip is null || latest.Index > tip.Index)
             {
-                _logger.Debug("Trying to fill up previous blocks...");
-                BlockChain<T> previousBlocks = await SyncPreviousBlocksAsync(
-                    _blockChain,
-                    peer,
-                    oldest.PreviousHash,
-                    null,
-                    blocks.Count,
-                    evaluateActions: true,
-                    cancellationToken: cancellationToken
-                );
-                _logger.Debug("Filled up. trying to concatenation...");
-
-                foreach (Block<T> block in blocks)
+                BlockChain<T> previousBlocks = null;
+                try
                 {
-                    previousBlocks.Append(block);
-                }
+                    _logger.Debug("Trying to fill up previous blocks...");
+                    previousBlocks = await SyncPreviousBlocksAsync(
+                        _blockChain,
+                        peer,
+                        oldest.PreviousHash,
+                        null,
+                        blocks.Count,
+                        evaluateActions: true,
+                        cancellationToken: cancellationToken
+                    );
+                    _logger.Debug("Filled up. trying to concatenation...");
 
-                _logger.Debug("Sync is done.");
-                if (!previousBlocks.Id.Equals(_blockChain.Id))
+                    foreach (Block<T> block in blocks)
+                    {
+                        previousBlocks.Append(block);
+                    }
+
+                    _logger.Debug("Sync is done.");
+                    if (!previousBlocks.Id.Equals(_blockChain.Id))
+                    {
+                        _logger.Debug("trying to swapping chain...");
+                        _blockChain.Swap(previousBlocks, render: true);
+                        _logger.Debug("Swapping complete");
+                    }
+                }
+                catch (Exception)
```

Comment:
How about converting this block to `finally`?

Reply:
Changed!